### PR TITLE
Find bucket grants

### PR DESCRIPTION
**Pull request by ian-mi**
Adds find bucket grants functionality for returning all grants for the given bucket of the specified type. Role names are returned as strings using a new bin2name utility function.

A match_grants helper function was also added to simplify several metadata folds.

### CODE REVIEW

**Comment by JeetKunDoug** [comment]
+1 - Thumbs seems confused here though - will retry and see if it fixes itself

**Comment by JeetKunDoug** [comment]
@thumbot retry

File: test/riak_core_security_tests.erl [hunk 1]
```diff
@@ -245,6 +245,23 @@ security_test_() ->
...
+    ok = riak_core_security:add_group("testgroup", []),
+    ok = riak_core_security:add_user("testuser1", [{groups, "testgroup"}]),
+    ok = riak_core_security:add_user("testuser2", []),
+    ok = riak_core_security:add_grant(["testuser1", "testuser2"], <<"bucket">>, ["riak_kv.get"]),
+    ok = riak_core_security:add_grant(["testuser2"], <<"bucket">>, ["riak_kv.put"]),
+    ok = riak_core_security:add_grant(all, <<"bucket">>, ["riak_kv.get"]),
+    ok = riak_core_security:add_grant(["group/testgroup"], <<"bucket">>, ["riak_kv.put"]),
+    Grants = riak_core_security:find_bucket_grants(<<"bucket">>, user),
+    GroupGrants = riak_core_security:find_bucket_grants(<<"bucket">>, group),
+    io:format("grants: ~w~n", [Grants]),
```
Comment (fadushin) [inline test/riak_core_security_tests.erl]: Stray debug message?

Comment (ian-mi) [reply test/riak_core_security_tests.erl]: Fixed

**Comment by fadushin** [comment]
+1

**Comment by fadushin** [comment]
Nicely done!

**Pull request merged**